### PR TITLE
oci-containers: consolidate capabilities interface

### PR DESCRIPTION
**Pull request by Yethal**
`capAdd` and `capDrop` were consolidated into a single `capabilities` option.
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/virtualisation/oci-containers.nix [hunk 1]
```diff
@@ -295,28 +295,19 @@ let
...
-
-        capDrop = mkOption {
-          type = with types; lazyAttrsOf (nullOr bool);
-          default = { };
-          description = ''
-            Capabilities to drop from container
+            Capabilities to configure for the container.
+            When set to true, capability is added to the container.
+            When set to false, capability is dropped from the container.
+            When null default runtime settings apply.
```
Comment (benley) [inline nixos/modules/virtualisation/oci-containers.nix:305]:
```suggestion
            When null, default runtime settings apply.
```

**Comment by benley** [comment]
I was initially opposed to this approach but now that I see it implemented it actually looks pretty reasonable.  Let's see what others might have to say about it.

**Comment by Yethal** [comment]
@benley Can we backport this to 24.11?

**Review by roberth — APPROVED** [review]
This matches what I had in mind in our [discussion](https://github.com/NixOS/nixpkgs/pull/349537#pullrequestreview-2486922190).
The behavior seems to be well covered by the tests, the documentation is clear, and it now supports overriding.
It also matches what Arion does (a docker compose wrapper), which is nice, and also means that it's a proven solution.

**Review by benley — APPROVED** [review]

**Comment by roberth** [comment]
> @benley @roberth Can we backport this to 24.11?

It's a grey area, but I'd support it for these reasons
- these two PRs are low risk
- backporting these changes makes future backports of fixes easier

**Pull request merged**